### PR TITLE
release notes edits for 0.12.0

### PR DESCRIPTION
**Pull request by rhero**
edits to release notes

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/808)

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -42,21 +42,20 @@ v0.12.0
          - Change
 
     *    - |breaking|
-         - If you do not specify a leader block in your config, AtlasDB will now still try to register the timestamp and lock endpoints necessary for other clients or CLIs to run in the same keyspace.
-           This may require changes in setup logic for applications that have previously only ever run with no leader block.
+         - AtlasDB will always try to register timestamp and lock endpoints for your application, whereas previously this only occurs if you specify a :ref:`leader-config`.
+           This ensures that CLIs will always be able to run against your service.
+           Applications that have previously only ever run with no leader block may need to make changes to startup logic.
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: I remember some discussion about this, but details escape me - what's the scenario where this is actually breaking? I'd like to include more details about who's affected.

I think @tpetracca went and actually tried this change on various products while trying out the backup stuff - did any of the products fail to startup?

File: docs/source/release_notes/release-notes.rst [hunk 2]
```diff
@@ -221,7 +220,7 @@ v0.9.0
          - Inserting an empty (size = 0) value into a ``Cell`` will now throw an ``IllegalArgumentException``. (`#156 <https://github.com/palantir/atlasdb/issues/156>`__) Likely empty
            values include empty strings and empty protobufs.
 
-           Atlas cannot currently distinguish between empty and deleted cells. In previous versions of Atlas, inserting
+           AtlasDB cannot currently distinguish between empty and deleted cells. In previous versions of Atlas, inserting
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: You missed another Atlas on this line

**Comment by gsheasby** [comment]
One nit, and of course the outstanding question, but otherwise looks fine.

**Comment by rhero** [comment]
ah, that's what I get for doing ctrl f on "atlas ".

File: docs/source/release_notes/release-notes.rst [hunk 3]
```diff
@@ -42,23 +42,25 @@ v0.12.0
          - Change
 
     *    - |breaking|
-         - If you do not specify a leader block in your config, AtlasDB will now still try to register the timestamp and lock endpoints necessary for other clients or CLIs to run in the same keyspace.
-           This may require changes in setup logic for applications that have previously only ever run with no leader block.
+         - AtlasDB will always try to register timestamp and lock endpoints for your application, whereas previously this only occurs if you specify a :ref:`leader-config`.
```
Comment (gbrova) [inline docs/source/release_notes/release-notes.rst]: Nit: this only _occurred_

File: docs/source/release_notes/release-notes.rst [hunk 4]
```diff
@@ -293,7 +295,7 @@ v0.7.0
          - Change
 
     *    - |new|
-         - Atlas can now be backed by Postgres via DB KVS. This is a very early release for this feature, so please contact us if you
+         - AtlasDB can now be backed by Postgres via DB KVS. This is a very early release for this feature, so please contact us if you
            plan on using it. Please see `the documentation <http://palantir.github.io/atlasdb/html/configuration/postgres_key_value_service_config.html>`_ for more details.
```
Comment (gbrova) [inline docs/source/release_notes/release-notes.rst]: AtlasDB docs links should be relative, not absolute.

Comment (gbrova) [inline docs/source/release_notes/release-notes.rst]: Also, do we want to add a PR link to this item?

**Pull request merged**